cypress/e2e/EngageSphere.cy.js

```diff
@@ -32,64 +32,268 @@ describe("EngageSphere", () => {
   });
 
   it("It renders the footer with the right text and links", () => {
-    // Arrange
-    const footerContainer = 'footer[class^="Footer_container"]';
-
     // Assert
-    cy.contains(footerContainer, "Copyright 2025 - Talking About Testing").should("be.visible");
+    cy.contains("footer", "Copyright 2025 - Talking About Testing").should("be.visible");
 
-    cy.contains(`${footerContainer} a`, "Blog")
+    cy.contains("footer a", "Blog")
       .should("be.visible")
       .and("have.attr", "href", "https://talkingabouttesting.com");
 
-    cy.contains(`${footerContainer} a`, "Courses")
+    cy.contains("footer a", "Courses")
       .should("be.visible")
       .and("have.attr", "href", "https://talking-about-testing.vercel.app/");
 
-    cy.contains(`${footerContainer} a`, "Podcast")
+    cy.contains("footer a", "Podcast")
       .should("be.visible")
       .and("have.attr", "href", "https://open.spotify.com/show/5HFlqWkk6qtgJquUixyuKo");
 
-    cy.contains(`${footerContainer} a`, "YouTube")
+    cy.contains("footer a", "YouTube")
       .should("be.visible")
       .and("have.attr", "href", "https://youtube.com/@talkingabouttesting");
   });
 
-  it("It shows the cookie banner on first visit", () => {
+  it("It renders the 'Hi there' greeting when no name is provided", () => {
+    // Assert
+    cy.get('[data-testid="name"]').should("be.empty");
+    cy.contains("h2", "Hi there").should("be.visible");
+  });
+
+  it("It renders the 'Hi Joe' greeting when name is provided", () => {
+    // Act
+    cy.get('[data-testid="name"]').type("Joe");
+    // Assert
+    cy.contains("h2", "Hi Joe").should("be.visible");
+  });
+
+  it("It renders the header with a heading, theme's toggle, and a text input field", () => {
+    // Assert
+    cy.get("[class^='Header_container']").should("be.visible").and("contain.text", "EngageSphere");
```

Comment:
Uma versão simplificada seria a seguinte:

```suggestion
    cy.contains("h1", "EngageSphere").should("be.visible");
```